ui/app/AppLayouts/Chat/ChatColumn/MessageComponents/UsernameLabel.qml

```diff
@@ -2,24 +2,62 @@ import QtQuick 2.3
 import "../../../../../shared"
 import "../../../../../imports"
 
-StyledTextEdit {
-    id: chatName
+Item {
+    id: root
+    height: childrenRect.height
+    width: chatName.width + ensOrAlias.width + ensOrAlias.anchors.leftMargin
+    property string userName: ""
+    property string localName: ""
+    property string alias: ""
+    property alias label: chatName
     visible: isMessage && authorCurrentMsg != authorPrevMsg
-    height: this.visible ? 18 : 0
-    //% "You"
-    text: !isCurrentUser ? Utils.removeStatusEns(userName) : qsTrId("You")
-    color: (userName.startsWith("@") || isCurrentUser) ? Style.current.blue : Style.current.textColor
-    font.bold: true
-    font.pixelSize: Style.current.secondaryTextFontSize
-    readOnly: true
-    wrapMode: Text.WordWrap
-    selectByMouse: true
-    MouseArea {
-        cursorShape: Qt.PointingHandCursor
-        acceptedButtons: Qt.LeftButton | Qt.RightButton
-        anchors.fill: parent
-        onClicked: {
-            clickMessage(true)
+
+    StyledTextEdit {
+        id: chatName
+        text: {
+            if (isCurrentUser) {
+                return qsTr("You")
+            }
+
+            if (root.localName !== "") {
+                return root.localName
+            }
+
+            if (root.userName !== "") {
+                return Utils.removeStatusEns(root.userName)
+            }
+            return Utils.removeStatusEns(root.alias)
```

Comment:
Because there's a possibility that `alias` holds the fully qualified ENS name, we gotta check for removal here as well..